src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -284,7 +284,17 @@ protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder
             {
                 diagnostics.Add(ErrorCode.WRN_NullAsNonNullable, parameterSyntax.Default.Value.Location);
             }
-
+            if (this.IsNullChecked)
```

Comment:
>if (this.IsNullChecked) [](start = 12, length = 23)

Can we combine this with the `if` above? It looks both blocks perform similar checks, but differently.